skbio/io/format/genbank.py

```diff
@@ -0,0 +1,669 @@
+
+
+@genbank.reader(Sequence)
+def _genbank_to_biological_sequence(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), seq_num)
+    return _construct(record, Sequence, **kwargs)
+
+
+@genbank.reader(DNA)
+def _genbank_to_dna(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), seq_num)
+    return _construct(record, DNA, **kwargs)
+
+
+@genbank.reader(RNA)
+def _genbank_to_rna(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), seq_num)
+    return _construct(record, RNA, **kwargs)
+
+
+@genbank.reader(Protein)
+def _genbank_to_protein(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), seq_num)
+    return _construct(record, Protein, **kwargs)
+
+
+@genbank.writer(None)
+def _generator_to_genbank(obj, fh, lowercase=None):
+    for obj_i in obj:
+        _serialize_single_genbank(obj_i, fh, lowercase)
+
+
+@genbank.writer(Sequence)
+def _biological_sequence_to_genbank(obj, fh, lowercase=None):
+    _serialize_single_genbank(obj, fh, lowercase)
+
+
+@genbank.writer(DNA)
+def _dna_to_genbank(obj, fh, lowercase=None):
+    _serialize_single_genbank(obj, fh, lowercase)
+
+
+@genbank.writer(RNA)
+def _rna_to_genbank(obj, fh, lowercase=None):
+    _serialize_single_genbank(obj, fh, lowercase)
+
+
+@genbank.writer(Protein)
+def _protein_to_genbank(obj, fh, lowercase=None):
+    _serialize_single_genbank(obj, fh, lowercase)
+
+
+def _construct(record, constructor=None, **kwargs):
+    seq, md, pmd = record
+    if 'lowercase' not in kwargs:
+        kwargs['lowercase'] = True
+    if constructor is None:
+        unit = md['LOCUS']['unit']
+        if unit == 'bp':
+            is_rna = 'u' in seq or 'U' in seq
+            # often RNA mol type has T instead of U for genbank from from NCBI;
+            # but in case of files from other sources...
+            constructor = RNA if is_rna else DNA
+        elif unit == 'aa':
+            constructor = Protein
+        else:
+            constructor = Sequence
+    return constructor(
+        seq, metadata=md, positional_metadata=pmd, **kwargs)
+
+
+def _parse_genbanks(fh):
+    data_chunks = []
+    for line in _line_generator(fh, skip_blanks=True, strip=False):
+        if line.startswith('//'):
+            yield _parse_single_genbank(data_chunks)
+            data_chunks = []
+        else:
+            data_chunks.append(line)
+
+
+def _parse_single_genbank(chunks):
+    metadata = {}
+    positional_metadata = None
+    sequence = ''
+    # each section starts with a HEADER without indent.
+    section_splitter = yield_section(lambda x: not x[0].isspace(), strip=False)
+    for section in section_splitter(chunks):
+        header = section[0].split(None, 1)[0]
+        parser = _PARSER_TABLE.get(
+            header, _parse_section_default)
+
+        if header == 'FEATURES':
+            parser = partial(
+                parser, length=metadata['LOCUS']['size'])
```

Comment:
Generally yes, it was more of a code organization comment. A future maintainer may not immediately grasp the dependency. So if it was refactored for the exceptional case (LOCUS) then it might be more obvious, it doesn't actually matter that much.